packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceTemplateQueryViewerStore.ts

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  extractElementNameFromPath,
+  type Query,
+  type RawLambda,
+} from '@finos/legend-graph';
+import {
+  QueryEditorStore,
+  type QueryPersistConfiguration,
+  type LegendQueryApplicationStore,
+  createViewProjectHandler,
+  createViewSDLCProjectHandler,
+} from '@finos/legend-application-query';
+import {
+  type DepotServerClient,
+  StoreProjectData,
+  LATEST_VERSION_ALIAS,
+} from '@finos/legend-server-depot';
+import { filterByType, guaranteeNonNullable, uuid } from '@finos/legend-shared';
+
+import { getDataSpace } from '../../graph-manager/DSL_DataSpace_GraphManagerHelper.js';
+import {
+  DataSpaceQueryBuilderState,
+  DataSpaceProjectInfo,
+} from './DataSpaceQueryBuilderState.js';
+import type { DataSpaceInfo } from './DataSpaceInfo.js';
+import { generateDataSpaceTemplateQueryViewerRoute } from '../../__lib__/query/DSL_DataSpace_LegendQueryNavigation.js';
+import {
+  DataSpaceExecutableTemplate,
+  type DataSpaceExecutionContext,
+} from '../../graph/metamodel/pure/model/packageableElements/dataSpace/DSL_DataSpace_DataSpace.js';
+import type { QueryBuilderState } from '@finos/legend-query-builder';
+import type { ProjectGAVCoordinates } from '@finos/legend-storage';
+import { DSL_DataSpace_getGraphManagerExtension } from '../../graph-manager/protocol/pure/DSL_DataSpace_PureGraphManagerExtension.js';
+import { retrieveAnalyticsResultCache } from '../../graph-manager/action/analytics/DataSpaceAnalysisHelper.js';
+import {
+  createQueryDataSpaceTaggedValue,
+  createQueryClassTaggedValue,
+} from './DataSpaceQueryCreatorStore.js';
+
+export class DataSpaceTemplateQueryViewerStore extends QueryEditorStore {
```

Comment:
should this be a creator store. does it make sense to abstract this and `DataSpaceQueryCreatorStore` or no?

Reply:
For template query, it's not `create` a template query but more about opening and editing it, therefore I named it a `viewer`. However, I renamed it to `DataSpaceTemplateQueryCreatorStore`.


I don't think we need to abstract it and `DataSpaceQueryCreatorStore ` again since QueryEditorStore is already the base class, and all methods in `DataSpaceTemplateQueryViewerStore ` and `DataSpaceQueryCreatorStore` are inherited from `QueryEditorStore`.

Besides, there are very few overlaps between `DataSpaceTemplateQueryViewerStore ` and `DataSpaceQueryCreatorStore`.  No common concrete function could be abstracted into a new abstract class. 👀😅